### PR TITLE
Add tooltip/ariaLabel for New Button if user has permissions

### PR DESCRIPTION
**Pull request by pascalwengerter**
## Description
Add tooltip/ariaLabel for New Button if user has permissions to upload/create resources

### CODE REVIEW

File: packages/web-app-files/src/components/AppBar.vue [hunk 1]
```diff
@@ -147,14 +145,14 @@ export default {
...
-    _cannotCreateDialogText() {
+    newButtonDescription() {
       if (!this.canUpload) {
         return this.$gettext('You have no permission to upload!')
       }
       if (!this.hasFreeSpace) {
         return this.$gettext('You have not enough space left to upload!')
       }
-      return null
+      return "Add files or folders"
```
Comment (kulmann) [inline packages/web-app-files/src/components/AppBar.vue:155]: This now adds a tooltip for a non-error case as well. IMO not good. Looks like it's required to separate into two different computed props, one for the tooltip and one for the aria-label.

Comment (pascalwengerter) [reply packages/web-app-files/src/components/AppBar.vue]: Please check the solution I went for and let me know if you like it!

File: packages/web-app-files/src/components/AppBar.vue [hunk 2]
```diff
@@ -306,6 +300,19 @@ export default {
...
+      if (!this.canUpload) {
+        return this.$gettext('You have no permission to upload!')
+      }
+      if (!this.hasFreeSpace) {
+        return this.$gettext('You have not enough space left to upload!')
+      }
+      if (isToolTip === true) {
+        return null
+      }
+      return 'Add files or folders'
```
Comment (kulmann) [inline packages/web-app-files/src/components/AppBar.vue:313]: This line lacks `this.$gettext`

File: packages/web-app-files/src/components/AppBar.vue [hunk 3]
```diff
@@ -146,15 +144,11 @@ export default {
...
-    _cannotCreateDialogText() {
-      if (!this.canUpload) {
-        return this.$gettext('You have no permission to upload!')
-      }
-      if (!this.hasFreeSpace) {
-        return this.$gettext('You have not enough space left to upload!')
-      }
-      return null
+    newButtonTooltip() {
+      return this.newButtonDescription(true)
```
Comment (kulmann) [inline packages/web-app-files/src/components/AppBar.vue:148]: `newButtonTooltip()` could do what we had before:
```js
if (!this.canUpload) {
         return this.$gettext('You have no permission to upload!')
       }
       if (!this.hasFreeSpace) {
         return this.$gettext('You have not enough space left to upload!')
       }
       return null
```

File: packages/web-app-files/src/components/AppBar.vue [hunk 4]
```diff
@@ -146,15 +144,11 @@ export default {
...
+    },
+    newButtonAriaLabel() {
```
Comment (kulmann) [inline packages/web-app-files/src/components/AppBar.vue:150]: ... and `newButtonAriaLabel()` could then do the following:
```js
const tooltip = this.newButtonTooltip
if (tooltip) {
    return tooltip
}
return this.$gettext('Add files or folders')
```

File: packages/web-app-files/src/components/AppBar.vue [hunk 5]
```diff
@@ -306,6 +300,19 @@ export default {
       this.createModal(modal)
     },
 
+    newButtonDescription(isToolTip) {
```
Comment (kulmann) [inline packages/web-app-files/src/components/AppBar.vue:303]: While this is totally fine technically, it is hard to read in the location where it's called. `this.newButtonDescription(true)` or `this.newButtonDescription(false)` doesn't give a hint what effect the boolean has on the output. I'd prefer a solution without the boolean, as pointed out in the two comments above.

**Review by kulmann — CHANGES_REQUESTED** [review]

**Review by kulmann — APPROVED** [review]
LGTM 🤖

**Pull request merged**